sdk/typescript/src/cryptography/keypair.ts

```diff
@@ -26,11 +29,17 @@ export abstract class BaseSigner {
 
 	async signWithIntent(bytes: Uint8Array, intent: IntentScope): Promise<SignedMessage> {
 		const intentMessage = messageWithIntent(intent, bytes);
-		const signature = await this.sign(intentMessage);
+		const digest = blake2b(intentMessage, { dkLen: 32 });
+
+		const signature = toSerializedSignature({
+			signature: await this.sign(digest),
+			signatureScheme: this.getKeyScheme(),
+			pubKey: this.getPublicKey(),
+		});
 
 		return {
-			bytes: intentMessage,
 			signature,
+			bytes,
```

Comment:
@Jordan-Mysten this is probably one of the changes I need more eyes on. 

The rawSigner flow was something like:

rawSigner.signData would create the digest, and call keypair.signData with the digest, then create the serializedSignature with the result.

The new methods on transactions never worked, because they were missing the digest piece (or I misunderstood how they were expected to be used).  Adding the digest to signWIthIntent seems like a good option, because signData/sign to me implies signing the data with the private key directly without a digest step, I think adding the digest in withIntent makes the most sense, but I might not fully understand this yet.

The other piece here is the returned bytes.  This was returning the intentMessage bytes, but I think thats slightly confusing.

When you use this to sign and execute a transaction this feels right:

```ts
const {signature, bytes } = keypair.signTransaction(txb.build())

client.executeTransaction({ transactionBlock: bytes, signature })
```

But this wouldn't work since the bytes are for the intent, not the transactions.  The updated version returns the passed in bytes, which looks a bit redundant at first, but I think makes more sense if you build directly in the arguments as shown above.

Reply:
One other question here:  all the methods are named like `signTransaction` rather than `signTransactionBlock` should we use the `transactionBlock` terminology more consistently for the new classes?

Reply:
I think the issue with this is that signMessage (currently) signs the message bytes directly right?

Reply:
Oh just kidding we currently hash everything we sign. I agree adding this to the intent is the right place, and I agree that `sign` should just be bytes in bytes out (this makes it easiest to re-implement keypairs without needing to re-implement anything else, because you don't need to care about formatting input or anything, you just sign)

And yeah these should all call them `TransactionBlocks` not transactions. We should never call the outer thing a `transaction`.